### PR TITLE
DGL Enter

### PR DESCRIPTION
**Pull request by VoVAllen**
## Description

Install dgl enter on this branch by 
`pip install git+https://github.com/VoVAllen/dgl.git@enter_merge#subdirectory=enter`

Need nightly version dgl for negative sampler

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: enter/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+The entry point of enter is `dgl-enter`, and it has three subcommand `config`, `train` and `export`.
+
+How to train a model on a dataset with task:
+1. Run `dgl-enter config` to generate a configuration file
+2. Run `dgl-enter train` to start a training script base on the configuration file generated. Or you can use `dgl-enter export` to export a self-contained, runnable python script based on your configuration file
+
+
+
+
+`dgl-ente` currently provides 4 pipelines:
```
Comment (yzh119) [inline enter/README.md:20]:
```suggestion
`dgl-enter` currently provides 4 pipelines:
```

**Review by yzh119 — CHANGES_REQUESTED** [review]

**Review by yzh119 — APPROVED** [review]

File: enter/dglenter/pipeline/graphpred/gen.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+class NodepredPipelineCfg(BaseModel):
+    node_embed_size: Optional[int] = -1
+    early_stop: Optional[EarlyStopConfig] = EarlyStopConfig()
+    num_epochs: int = 200
+    eval_period: int = 5
+    optimizer: dict = {"name": "Adam", "lr": 0.005}
+    loss: str = "CrossEntropyLoss"
+@PipelineFactory.register("nodepred")
+class NodepredPipeline(PipelineBase):
```
Comment (jermainewang) [inline enter/dglenter/pipeline/graphpred/gen.py:22]: Consider delete the entire graphpred folder as we are cutting it off. Remember to save it for later develop.

File: enter/dglenter/pipeline/linkpred/gen.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+from ...utils.factory import PipelineFactory, NodeModelFactory, PipelineBase, DataFactory, EdgeModelFactory, NegativeSamplerFactory
+from ...utils.base_model import EarlyStopConfig, DeviceEnum
+
+from ...utils.yaml_dump import deep_convert_dict, merge_comment
+import ruamel.yaml
+from ruamel.yaml.comments import CommentedMap
+
+class LinkpredPipelineCfg(BaseModel):
+    hidden_size: int = 256
+    # early_stop: Optional[EarlyStopConfig] = EarlyStopConfig()
```
Comment (jermainewang) [inline enter/dglenter/pipeline/linkpred/gen.py:17]: If it's not supported. Remove the option.

**Review by jermainewang — APPROVED** [review]

File: enter/dglenter/utils/early_stop.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import torch
+
+class EarlyStopping:
```
[the remark this answers is not in the document]
Comment (VoVAllen) [reply enter/dglenter/utils/early_stop.py:3]: Will do this in a seperate PR

**Pull request merged**